tests/test_backend.py

```diff
@@ -810,17 +810,61 @@ def test_clear(self, cache: RedisCache):
         assert value_from_cache_after_clear is None
 
     def test_hset(self, cache: RedisCache):
-        if isinstance(cache.client, ShardClient):
-            pytest.skip("ShardClient doesn't support get_client")
         cache.hset("foo_hash1", "foo1", "bar1")
         cache.hset("foo_hash1", "foo2", "bar2")
         assert cache.hlen("foo_hash1") == 2
         assert cache.hexists("foo_hash1", "foo1")
         assert cache.hexists("foo_hash1", "foo2")
 
+    def test_hsetnx(self, cache: RedisCache):
+        result_foo1 = cache.hsetnx("foo_hash1", "foo1", "bar1")
+        result_foo2 = cache.hsetnx("foo_hash1", "foo2", "bar2")
+        result_foo2_1 = cache.hsetnx("foo_hash1", "foo2", "bar2")
```

Comment:
could you change the value written?
```suggestion
        result_foo2_1 = cache.hsetnx("foo_hash1", "foo2", "bar3")
```